### PR TITLE
feat: `override_timestamps_with_wall_time` parameter

### LINKED ISSUE #546: Support for `use_sim_time` = `false`
What are your thought about running the bridge with `use_sim_time` `false`? Would you be open for a PR like this: https://github.com/gazebosim/ros_gz/pull/545

Please give me your thoughts.

Thanks!

### PR DESCRIPTION
**Pull request by reinzor**
# 🎉 New feature

Closes #546 

## Summary
Stamp all outgoing headers with the wall time if parameter `override_timestamps_with_wall_time` is set to `true`.

## Test it
Start the bridge with the additional parameter `override_timestamps_with_wall_time` set to `true`. Note that the outgoing messages have the wall time stamp and not the gazebo time stamp.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
code is not compiling https://github.com/gazebosim/ros_gz/actions/runs/9579762345/job/26418501612?pr=562

**Comment by azeey** [comment]
@Mergifyio update

**Comment by azeey** [comment]
CI issues should hopefully be fixed after https://github.com/ros2/ros2/pull/1574

**Comment by Timple** [comment]
Let's re-run them then :slightly_smiling_face:

**Comment by azeey** [comment]
I just reran it, but I realized the branch is out-of-date. Do you mind merging from `ros2`?

**Comment by reinzor** [comment]
> I just reran it, but I realized the branch is out-of-date. Do you mind merging from `ros2`?

done

**Comment by azeey** [comment]
It seems like there are some CI failures. Do the tests pass locally for you?

**Comment by reinzor** [comment]
> It seems like there are some CI failures. Do the tests pass locally for you?

I am running the tests locally and they seem to fail as well. However, I don't see how this code change would affect the failing tests. Could you help me with this?

**Comment by azeey** [comment]
The error logs indicated that it was failing to create any of the GZ->ROS bridges. Inspecting the exception with a print statement shows that the problem is that we are redeclaring the `override_timestamps_with_wall_time` parameter. My suggestion is to declare the parameter in https://github.com/gazebosim/ros_gz/blob/4c64bbea8041131b36a84692af7064a4abbfddb4/ros_gz_bridge/src/ros_gz_bridge.cpp#L35 and use `get_parameter` to actually retrieve the value. I would also suggest adding a member variable to `BridgeHandle` that stores the value of the parameter and fetch the parameter in the constructor of `BridgeHandle` instead of modifying the function signature.

**Comment by reinzor** [comment]
Thanks for your help @azeey ; I updated the PR with the suggestions you mentioned. Not sure about not changing the function signature though. We need to pass this variable to the factory right?

File: ros_gz_bridge/src/factory.hpp
Comment (ahcorde) [inline ros_gz_bridge/src/factory.hpp:34]: include `<type_traits>`

File: ros_gz_bridge/src/factory.hpp
Comment (ahcorde) [inline ros_gz_bridge/src/factory.hpp:162]: include `<chrono>`

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
> Thanks for your help @azeey ; I updated the PR with the suggestions you mentioned. Not sure about not changing the function signature though. We need to pass this variable to the factory right?

You are targeting `ros2`, we can change the API's here, but if we need to backport this, we should do it in a different way

**Comment by reinzor** [comment]
> > Thanks for your help @azeey ; I updated the PR with the suggestions you mentioned. Not sure about not changing the function signature though. We need to pass this variable to the factory right?
> 
> You are targeting `ros2`, we can change the API's here, but if we need to backport this, we should do it in a different way

I will leave that up to you as maintainers. Feel free to move the logic to a different place.

**Review by ahcorde — APPROVED** [review]

**Review by azeey — APPROVED** [review]
LGTM! Thanks for iterating. I realized the API change issue is not a problem since none of the changed headers are considered public API. I think this could actually be backported if needed.

**Comment by reinzor** [comment]
Great thanks! How does this backporting to `jazzy` work; should I just send another PR that cherry-picks [this commit](https://github.com/gazebosim/ros_gz/commit/6daea2caf9f0190f83ec7a73e7ed13307b193029)?

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**
